src/OSPSuite.Infrastructure.Import/Core/IDataSource.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Linq.Dynamic;
+using System.Reflection;
+using OSPSuite.Core.Import;
+using OSPSuite.Infrastructure.Import.Services;
+using OSPSuite.Utility.Collections;
+
+namespace OSPSuite.Infrastructure.Import.Core
+{
+   /// <summary>
+   /// Collection of DataSets
+   /// </summary>
+   public interface IDataSource
+   {
+      void SetDataFormat(IDataFormat dataFormat);
+      void SetNamingConvention(string namingConvention);
+      void AddSheets(Cache<string, IDataSheet> dataSheets, IReadOnlyList<ColumnInfo> columnInfos, string filter);
+      void SetMappings(string fileName, IEnumerable<MetaDataMappingConverter> mappings);
+      ImporterConfiguration GetImporterConfiguration();
+      IEnumerable<MetaDataMappingConverter> GetMappings();
+      Cache<string, IDataSet> DataSets { get; }
+      IEnumerable<string> NamesFromConvention();
+      NanSettings NanSettings { get; set; }
+   }
+
+   public class DataSource : IDataSource
+   {
+      private IImporter _importer;
+      private ImporterConfiguration _configuration;
+
+      public DataSource(IImporter importer)
+      {
+         _importer = importer;
+         _configuration = new ImporterConfiguration();
+
+         var type = typeof(DynamicQueryable).Assembly.GetType("System.Linq.Dynamic.ExpressionParser");
```

Comment:
What is this?

Reply:
Old attempt to start with the filters... already removed